great_expectations/rule_based_profiler/domain_builder/categorical_column_domain_builder.py

```diff
@@ -0,0 +1,247 @@
+from typing import Dict, List, Optional, Union
+
+from great_expectations.core.batch import BatchRequest, RuntimeBatchRequest
+from great_expectations.exceptions import ProfilerConfigurationError
+from great_expectations.execution_engine.execution_engine import MetricDomainTypes
+from great_expectations.rule_based_profiler.domain_builder import DomainBuilder
+from great_expectations.rule_based_profiler.domain_builder.cardinality_checker import (
+    CardinalityChecker,
+    CardinalityLimitMode,
+)
+from great_expectations.rule_based_profiler.domain_builder.domain_builder import (
+    build_domains_from_column_names,
+)
+from great_expectations.rule_based_profiler.types import Domain, ParameterContainer
+from great_expectations.validator.metric_configuration import MetricConfiguration
+from great_expectations.validator.validator import Validator
+
+
+class CategoricalColumnDomainBuilder(DomainBuilder):
+    """
+    This DomainBuilder uses column cardinality to identify domains.
+    """
+
+    def __init__(
+        self,
+        data_context: "DataContext",  # noqa: F821
+        batch_request: Optional[Union[BatchRequest, RuntimeBatchRequest, dict]] = None,
```

Comment:
@anthonyburdi Constructor must also take in `batch: Optional[Batch] = None,` as an argument (please see `ColumnDomainBuilder`) -- this is important for Self-Initializing Expectations.

Reply:
@alexsherstinsky What is the purpose of this for the CategoricalColumnDomainBuilder? I see that it is used in the proposed `_validate_rule_and_update_rule_properties` but the `ValueSetMultiBatchParameterBuilder` builds its own set of parameters from all batches provided (not just the active batch).

Reply:
@anthonyburdi This is important for Self-Initializing Expectations whereby there is no batch_request, just Batch or Batch list.  Happy to discuss.  Thanks.

Reply:
@anthonyburdi and @alexsherstinsky discussed and agreed to add capabilities needed to enable Self-Initializing expectations that use ValueSet in a separate work item.